src/components/searchresultsitems.main.jsx

```diff
@@ -115,45 +114,39 @@ class SearchResultsItemsMain extends React.Component {
   render() {
     const { searchKeywordsProps } = this.props;
     const { searchResultsModel, searchKeywords } = this.state;
-    if (searchResultsModel.links.length > 0 && searchKeywords === searchKeywordsProps) {
+    const isLoading = searchResultsModel.links.length === 0 || searchKeywords !== searchKeywordsProps;
+
+    if (isLoading) {
       return (
-        <div className="category-items-container container">
-          <div data-region="categoryTitleRegion" style={{ display: 'block' }}>
-            <div>
-              <h1 className="view-title">
-                {intl.get('search-results')}
-              </h1>
-            </div>
-          </div>
-          <ProductListPaginationTop paginationDataProps={searchResultsModel._items ? searchResultsModel._items[0] : searchResultsModel} />
-          <ProductListMain productData={searchResultsModel._items ? searchResultsModel._items[0] : searchResultsModel} />
-          <ProductListPaginationBottom paginationDataProps={searchResultsModel._items ? searchResultsModel._items[0] : searchResultsModel} />
-        </div>
+        <div className="loader" />
       );
     }
-    if (searchResultsModel.links.length === 0 && searchResultsModel.pagination && searchKeywords === searchKeywordsProps) {
-      return (
-        <div className="category-items-container container">
-          <div data-region="categoryTitleRegion" style={{ display: 'block' }}>
+
+    const products = searchResultsModel._items ? searchResultsModel._items[0] : searchResultsModel;
+    const noProducts = !products || products.links.length === 0;
```

Comment:
You can run into a case where `isLoading ` is true, and you end up returning the loader before even being able to check if you have any products on the page. That's why originally we had the loader as the last fallback rather than the first. The loader is also shown if a user is offline and loads the page.

ex. search for `abc`, the loader just keeps spinning. 😢

Reply:
Yeah, it keeps spinning and spinning  🤪

Reply:
Fixed, take a look at the screen capture in another comment.